runtime/appruntime/apisdk/api/errmarshalling/jsonextension.go

```diff
@@ -0,0 +1,159 @@
+package errmarshalling
+
+import (
+	"fmt"
+	"reflect"
+	"unsafe"
+
+	jsoniter "github.com/json-iterator/go"
+	"github.com/modern-go/reflect2"
+)
+
+var (
+	// jsonMarshaller by reflect2 type (we use this for speed in the extension)
+	serdeByType = map[reflect2.Type]*jsonMarshaller{}
+
+	// jsonMarshaller by name (we use this for use in Unmarshal)
+	serdeByName = map[string]*jsonMarshaller{}
+
+	// errType is the reflect2 type of the error interface
+	errType = reflect2.Type2(reflect.TypeOf((*error)(nil)).Elem())
+)
+
+const (
+	// TypeKey is the key used to identify the type of an error
+	// used to unmarshal the correct error type
+	TypeKey = "$type$"
```

Comment:
Maybe use `@type` to align with Protobuf's JSON mapping convention?

Reply:
Sure, I was following another convention that I thought was GraphQL, but in hindsight I'm not actually sure. Happy to change though!